clang/lib/StaticAnalyzer/Checkers/WebKit/UncountedLambdaCapturesChecker.cpp

```diff
@@ -106,6 +107,26 @@ class UncountedLambdaCapturesChecker
         return safeGetName(NsDecl) == "WTF" && safeGetName(Decl) == "switchOn";
       }
 
+      bool VisitCXXConstructExpr(CXXConstructExpr *CE) override {
+        if (ConstructToIgnore.contains(CE))
+          return true;
+        if (auto *Callee = CE->getConstructor()) {
+          unsigned ArgIndex = 0;
+          for (auto *Param : Callee->parameters()) {
+            if (ArgIndex >= CE->getNumArgs())
+              return true;
+            auto *Arg = CE->getArg(ArgIndex)->IgnoreParenCasts();
+            if (auto *L = findLambdaInArg(Arg)) {
```

Comment:
For my understanding: Is the LambdaExpr not covered in `VisitLambdaExpr`? What's special about the expression being an argument to a constructor?

Reply:
Oh, so we wanna ignore lambda expressions that are passed to `[[clang::noescape]]`. The code in `VisitCallExpr` and now `VisitCXXConstructExpr` is there to detect this specific case, and then ignore the argument.